metadata-ingestion/src/datahub/ingestion/source/s3/source.py

```diff
@@ -373,6 +373,35 @@ def read_file_spark(self, file: str, ext: str) -> Optional[DataFrame]:
         # see https://mungingdata.com/pyspark/avoid-dots-periods-column-names/
         return df.toDF(*(c.replace(".", "_") for c in df.columns))
 
+
+    def infer_file_extension(self, file, table_data):
+        fields = []
+        try:
+            fields = parquet.ParquetInferrer().infer_schema(file)
+        except ArrowInvalid as ai:
+            try:
+                fields = json.JsonInferrer().infer_schema(file)
+            except ValueError as ve:
+                try:
+                    fields = avro.AvroInferrer().infer_schema(file)
+                except TypeError as te:
+                    try:
+                        fields = csv_tsv.CsvInferrer(
+                            max_rows=self.source_config.max_rows
+                        ).infer_schema(file)
+                    except:
+                        try:
+                            fields = csv_tsv.TsvInferrer(
+                                max_rows=self.source_config.max_rows
+                            ).infer_schema(file)
+                        except:
+                            self.report.report_warning(
+                                table_data.full_path,
+                                f"file {table_data.full_path} has unsupported extension",
+                            )
```

Comment:
You don't really need this. You can let this exception bubble up and hit line 449 where a message will be logged.